src/services/TestResultsService.ts

```diff
@@ -27,71 +26,20 @@ export class TestResultsService {
     this.vehicleTestController.dataProvider.testResultsDAO = this.testResultsDAO;
   }
 
-  // TODO: CVSB-18780: refactor to guard logic, handle errors via Promise.reject and destructure object where needed
-  public async getTestResultBySystemNumber(filters: models.ITestResultFilters): Promise<any> {
-    if (filters.systemNumber && ValidationUtil.validateGetTestResultFilters(filters)) {
-      try {
-          const result = await this.testResultsDAO.getBySystemNumber(filters.systemNumber);
-          const response: models.ITestResultData = { Count: result.Count, Items: result.Items };
-          return this.applyTestResultsFilters(response, filters);
-        } catch (error) {
-          if (!(error instanceof models.HTTPError)) {
-            console.error("dynamoError on getBySystemNumber", error);
-            error = new models.HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR);
-          }
-          throw error;
-        }
-      } else {
-        // console.log("invalid systemNumber Filters", filters);
-        return Promise.reject(new models.HTTPError(400, enums.MESSAGES.BAD_REQUEST));
-      }
+  public async getTestResultBySystemNumber(filters: models.ITestResultFilters): Promise<models.ITestResult[]> {
```

Comment:
I wouldn't spend too much refactoring the old code in this service since it will be deprecated. Actually, shouldn't you mark these functions as deprecated?

Reply:
This function may still be used for VTM.